<?xml version="1.0"?>
<robot name="scalebot">
  <link name="base">
    <visual>
      <origin xyz="0.016 0 0.017" rpy="0 -0.1 0"/>
      <geometry>
        <mesh filename="package://scalebot/meshes/chair.stl"
          scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="white">
              <color rgba=".7 .65 .65 1"/>
      </material>

    </visual>
  </link>

  <link name="tracks">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://scalebot/meshes/tracks.stl"
          scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>

  <joint name="track_forks" type="continuous">
    <parent link="base"/>
    <child link="tracks"/>
    <axis xyz="0 1 0"/>
    <origin xyz="-.39 -0.0148 0" rpy="0 0 0"/>
  </joint>

  <link name="support">
    <visual>
      <origin xyz="0.025 0 .15" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://scalebot/meshes/support.stl"
          scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>

  <joint name="support_pivot" type="continuous">
    <parent link="base"/>
    <child link="support"/>
    <axis xyz="0 1 0"/>
    <origin xyz="-0.025 0 -0.15" rpy="0 0 0"/>
  </joint>

  <link name="laser_mount_link">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <box size="0 0 0"/>
        </geometry>

      </visual>
  </link>

  <joint name="base_to_laser_mount" type="continuous">
    <parent link="base"/>
    <child link="laser_mount_link"/>
    <axis xyz="0 1 0"/>
    <origin xyz=".233 0 .183" rpy="0 0 0"/>
  </joint>

</robot>
